src/ts/packedrtree.ts

```diff
@@ -171,7 +171,7 @@ export async function* streamSearch(
 
             if (isLeafNode) {
                 const featureLength = (() => {
-                    if (pos < numItems - 1) {
+                    if (pos < end - 1) {
```

Comment:
I don't think this change is correct. 

Here's my understanding:

`featureLength` the length, in bytes, of this feature in the feature buffer.

`numItems`: the number of leaf nodes in the tree. AKA the number of features in the dataset.

`pos`: the index of the feature in the feature set, which is the same as the ordering of the leaf nodes. (aka it's hilbert ordering relative to the other features in the feature set).

`end`: the index of the last `byte` in the batch of index nodes we're considering.

After perusing the index to figure out where our features live, we fetch the features from the feature buffer. For transfer efficiency and speed, we want to fetch "just enough" data.  Knowing each feature's `featureLength` is how we do that. 

With the PackedRTree index we can figure out the `featureLength` of every feature *except* for the very last feature in the entire dataset. That's what this branch is about - hence the comparison to `numItems`. 

It might be clearer to invert the condition and write it as:

```
console.assert(pos <= numItems, "can't have an index beyond the number of features");
const isLastFeatureInDataset = pos == end;
if (isLastFeatureInDataset) {
  // This is the last feature - there's no "next" feature
  // to measure to, so we can't know it's length.
  return null;
} else {
  // Since features are tightly packed, we infer the
  // length of _this_ feature by measuring to the _next_
  // feature's start.
  const nextPos = (pos - nodeIndex + 1) * 40;
  const nextOffset = dataView.getBigUint64(
    nextPos + 32,
    true,
  );
  return nextOffset - offset;
}
```

So anyway, I hope it's clear now it doesn't make sense to compare `pos` (the index of a feature) to `end` the index of a byte.

I'll take a look and see if I can get to the bottom of things... and maybe make a PR to clear up some of these variable names, which are currently not very helpful.

Reply:
Ok but as the buffer (that the dataView spans) is, depending on size etc, possibly only a subset of the nodes in a tree level it also doesn't seem right it can always peek the nextOffset up to the last feature. (i.e that would only be true of the full set of nodes is in the buffer?)